donazioni/viste.py

```diff
@@ -1,27 +1,194 @@
-from django.http import Http404
-from django.shortcuts import redirect
-from django.utils.decorators import method_decorator
-from django.views.generic import TemplateView
-
-from anagrafica.permessi.costanti import GESTIONE_CAMPAGNE, GESTIONE_CAMPAGNA, ERRORE_PERMESSI
-from autenticazione.funzioni import pagina_privata, VistaDecorata
-
-
-class DonazioniWelcomePage(VistaDecorata, TemplateView):
-    template_name = 'donazioni.html'
-
-    @method_decorator(pagina_privata(permessi=[GESTIONE_CAMPAGNE]))
-    def get(self, request, *args, **kwargs):
-        try:
-            return super(DonazioniWelcomePage, self).get(request, *args, **kwargs)
-        except Http404:
-            return redirect(ERRORE_PERMESSI)
-
-    def get_context_data(self, **kwargs):
-        contesto = super(DonazioniWelcomePage, self).get_context_data(**kwargs)
-        me = contesto.get('me')
-        contesto.update({
-            "sedi": me.oggetti_permesso(GESTIONE_CAMPAGNE),
-            "campagne": me.oggetti_permesso(GESTIONE_CAMPAGNA),
-        })
-        return contesto
+from datetime import datetime
+from itertools import chain
+
+from django.core.urlresolvers import reverse
+from django.db.models import Q
+from django.shortcuts import redirect, get_object_or_404
+
+from anagrafica.permessi.applicazioni import RESPONSABILE_CAMPAGNA
+from anagrafica.permessi.costanti import GESTIONE_CAMPAGNE, GESTIONE_CAMPAGNA, COMPLETO, ERRORE_PERMESSI, MODIFICA
+from autenticazione.funzioni import pagina_privata
+from donazioni.forms import ModuloCampagna, ModuloEtichetta
+from donazioni.models import Campagna, Etichetta
+
+
+@pagina_privata
+def donazioni_home(request, me):
+    contesto = {
+        'sedi': me.oggetti_permesso(GESTIONE_CAMPAGNE),
+        'campagne': me.oggetti_permesso(GESTIONE_CAMPAGNA),
+    }
+    template = 'donazioni.html'
+    return template, contesto
```

Comment:
Non un problema, solo coerenza stilistica -- puoi preferire il tenere la stringa del template inline? E.g. 
```python
return "donazioni.html", contesto
```
Eccezione chiaramente nel caso in cui la stringa sia composta (in quel caso fai la composizione in una altra riga).